src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2735,6 +2733,27 @@ def groupby_agg(
                 pandas_labels_for_columns_to_exclude=by_list,
             )
 
+        if agg_func in ("all", "any"):
+            empty_columns = (
+                len(
+                    [
+                        label
+                        for label in query_compiler._modin_frame.data_columns_index
+                        if label not in by_list
+                    ]
+                )
+                == 0
+            )
+            if not empty_columns and not all(
+                is_bool_dtype(t) or is_integer_dtype(t)
```

Comment:
Let's do not do specialized type check here to be consistent among all aggregations. we simply let the aggregation goes through, and let snowflake engine decides whether it is valid to apply the aggregation or not. One of the very typical use case is variant, if customer has a variant column with all numeric type, some aggregation will continue be valid there, furthermore, the variant column can be produced by snowpark pandas during some operaiton like apply

Reply:
Sure, I can remove it for now. The only concern I have is that in the future we might want to emit different SQL for non-numeric/boolean columns. Right now, we're using the `BOOLAND_AGG`/`BOOLOR_AGG` SQL functions on columns to support numbers/bools, but it's made clear in [documentation](https://docs.snowflake.com/en/sql-reference/functions/booland_agg#usage-notes) that these functions are not meant to be used with string types. Given that Python's notion of "truthiness" for strings is different from SQL's, if we want to support cases like `pd.Series(["abc", "", None]).all()` (which is at least moderately common in pandas), I think it would make sense to emit different SQL.